lib/puppet/defaults.rb

```diff
@@ -939,26 +945,37 @@ def self.default_checksum_types
 EOT
     },
     :digest_algorithm => {
-        :default  => lambda { default_digest_alg },
+        :default  => lambda { default_digest_algorithm },
         :type     => :enum,
-        :values => ["md5", "sha256", "sha384", "sha512", "sha224"],
-        :desc     => 'Which digest algorithm to use for file resources and the filebucket.
-                      Valid values are md5, sha256, sha384, sha512, sha224. Default is md5.',
+        :values   => valid_digest_algorithms,
+        :desc     => "Which digest algorithm to use for file resources and the filebucket.
+                      Valid values are #{valid_digest_algorithms.join(', ')}. Default is
+                      #{valid_digest_algorithms.first}.",
```

Comment:
Why is this `valid_digest_algorithms.first` instead of `default_digest_algorithm` (to match what's in the default lambda)?